src/PostgREST/App.hs

```diff
@@ -96,8 +95,8 @@ transactionMode structure target action =
     ActionInvoke{isReadOnly=False} ->
       let proc =
             case target of
-              (TargetProc qi) -> M.lookup (qiName qi) $
-                                   dbProcs structure
+              -- TODO: this proc search won't be accurate when the function is overloaded
```

Comment:
Although it's not perfect, you could improve accuracy by creating a set of the `pdVolatility` values for all matching functions, and then choose the minimal value in that set as defined by the Ord instance of `ProcVolatility`. (I think minimal is what I mean…basically Volatile if any are Volatile, going on up the list to Immutable if all are Immutable).

Reply:
I thought of leaving it as a TODO and see if later after a refactor I could pass the `PayloadJSON` keys so I can make the proc search accurate, thought it would make no harm since prior to this change the proc search also didn't consider the overloaded functions case.

But I think I can do as you said for now to somewhat improve the accuracy.

Reply:
On second thought it'll be better if I refactor this part now, that way the proc search is accurate.